django/db/models/query.py

```diff
@@ -223,6 +223,9 @@ def __or__(self, other):
         combined.query.combine(other.query, sql.OR)
         return combined
 
+    def __contains__(self, obj):
+        return isinstance(obj, self.model) and self.filter(pk=obj.pk).exists()
```

Comment:
This will subtly change existing code. First, if the queryset is already evaluated, then currently __contains__ checks against the cached results. Second, when calling x in qs, the queryset doesn't get evaluated any more.

My preference is to not change this unless there is large-scale support to change the behavior.